src/views/WelcomeView.vue

```diff
@@ -23,93 +15,76 @@ import NcLoadingIcon from '@nextcloud/vue/dist/Components/NcLoadingIcon.js'
 		return {
-			IconTalk,
+			name: t('spreed', 'Creating and joining a conversation with "{userid}"', { userid: callUser.value ?? '' }),
+			description: '',
 		}
-	},
+	}
 
-	data() {
+	if (callUser.value) {
 		return {
-			isCreatingConversationForCallUser: false,
-		}
-	},
-
-	computed: {
-		callUser() {
-			return this.$route.query.callUser
-		},
-
-		text() {
-			if (this.isCreatingConversationForCallUser) {
-				return {
-					name: t('spreed', 'Creating and joining a conversation with "{userid}"', { userid: this.callUser }),
-					description: '',
-				}
-			}
-
-			if (this.callUser) {
-				return {
-					name: t('spreed', 'Joining a conversation with "{userid}"', { userid: this.callUser }),
-					description: '',
-				}
-			}
-
-			return {
-				name: t('spreed', 'Join a conversation or start a new one'),
-				description: t('spreed', 'Say hi to your friends and colleagues!'),
-			}
-		}
-	},
-
-	watch: {
-		callUser: {
-			immediate: true,
-			handler() {
-				if (this.callUser) {
-					this.createAndJoinConversationForCallUser()
-				}
-			}
+			name: t('spreed', 'Joining a conversation with "{userid}"', { userid: callUser.value }),
+			description: '',
 		}
-	},
+	}
 
-	methods: {
-		t,
-		async createAndJoinConversationForCallUser() {
-			// Try to find an existing conversation
-			const conversation = this.$store.getters.getConversationForUser(this.callUser)
-			if (conversation) {
-				this.$router.push({
-					name: 'conversation',
-					params: { token: conversation.token },
-				})
-				return
-			}
+	return {
+		name: t('spreed', 'Join a conversation or start a new one'),
+		description: t('spreed', 'Say hi to your friends and colleagues!'),
+	}
+})
+
+const createAndJoinConversationForCallUser = async () => {
+	// Try to find an existing conversation
+	const conversation = store.getters.getConversationForUser(callUser.value)
+	if (conversation) {
+		Router.push({
+			name: 'conversation',
+			params: { token: conversation.token },
+		})
+		return
+	}
 
-			// Create a new one-to-one conversation
-			this.isCreatingConversationForCallUser = true
-			try {
-				const newConversation = await this.$store.dispatch('createOneToOneConversation', this.callUser)
-				this.$router.push({
-					name: 'conversation',
-					params: { token: newConversation.token },
-				})
-			} catch (error) {
-				showError(t('spreed', 'Error while joining the conversation'))
-				console.error(error)
-				this.$router.push({ name: 'notfound' })
-			}
-		},
+	// Create a new one-to-one conversation
+	isCreatingConversationForCallUser.value = true
+	try {
+		const newConversation = await store.dispatch('createOneToOneConversation', callUser.value)
+		Router.push({
+			name: 'conversation',
+			params: { token: newConversation.token },
+		})
+	} catch (error) {
+		showError(t('spreed', 'Error while joining the conversation'))
+		console.error(error)
+		Router.push({ name: 'notfound' })
 	}
 }
+
+watch(callUser, (newCallUser) => {
```

Comment:
RFC: instead, Router navigation guard can be used: https://router.vuejs.org/guide/advanced/composition-api.html#Navigation-Guards

This could  provide us more opportunities, e.g. prevent navigation and show input dialog like "You don't have a conversation with this person yet. Do you like to write your first message?" before creating